core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -148,7 +148,13 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** Cleanup local dirs and stop shuffle sender. */
   private[spark] def stop() {
     // Remove the shutdown hook.  It causes memory leaks if we leave it around.
-    Runtime.getRuntime.removeShutdownHook(shutdownHook)
+    if(shutdownHook != null) {
```

Comment:
I don't think it can be.  Unless you're calling `stop` before the constructor has finished, which doesn't seem possible.

The docs for `removeShutdownHook` imply that you can pass it any thread, and it'll just return false if that thread isn't currently registered, so even double-calls are unlikely to cause problems there.  There's just the IllegalStateException, which could happen if someone else's shutdown hook is calling stop.  That's definitely worth protecting against.